src/switch_rtp.c

```diff
@@ -5882,6 +5890,92 @@ static switch_size_t do_flush(switch_rtp_t *rtp_session, int force, switch_size_
+
+#ifdef DEBUG_SRTP_FLUSH
+                    if (sbytes) {
+                        q = 0;
+                        bytes_buf[0] = 0;
+                        for(i = 0; sbytes && i < sbytes; i++) {
+                            sprintf(bytes_buf + q, "0x%02x ", ((u_char *)(&rtp_session->recv_msg))[i]);
+                            q = strlen(bytes_buf);
+                        }
+                        switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_ERROR, "do_flush loop BEFORE decryption: bytes = %d\nData: %s\n", sbytes, bytes_buf);
+                    }
+#endif
+
+                    // Ensure we properly decrypt SRPT before reading DTMF
+                    if (!rtp_session->flags[SWITCH_RTP_FLAG_SECURE_RECV_MKI]) {
+                        stat = srtp_unprotect(rtp_session->recv_ctx[rtp_session->srtp_idx_rtp], &rtp_session->recv_msg.header, &sbytes);
+                    } else {
+                        stat = srtp_unprotect_mki(rtp_session->recv_ctx[rtp_session->srtp_idx_rtp], &rtp_session->recv_msg.header, &sbytes, 1);
+                    }
+
+#ifdef DEBUG_SRTP_FLUSH
+                    if (sbytes) {
+                        q = 0;
+                        bytes_buf[0] = 0;
+                        for(i = 0; sbytes && i < sbytes; i++) {
+                            sprintf(bytes_buf + q, "0x%02x ", ((u_char *)(&rtp_session->recv_msg))[i]);
+                            q = strlen(bytes_buf);
+                        }
+                        switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_ERROR, "do_flush loop AFTER decryption: bytes = %d\nData: %s\n", sbytes, bytes_buf);
+                    }
+#endif
+
+                    if (rtp_session->flags[SWITCH_RTP_FLAG_NACK] && stat == srtp_err_status_replay_fail) {
+                        /* false alarm nack */
+                        switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(rtp_session->session), SWITCH_LOG_DEBUG1, "REPLAY ERR, FALSE NACK\n");
+                        sbytes = 0;
+                        bytes = 0;
+                        if (rtp_session->stats.rtcp.pkt_count) {
+                            rtp_session->stats.rtcp.period_pkt_count--;
+                            rtp_session->stats.rtcp.pkt_count--;
+                        }
+                        goto done_srtp;
+                    }
+                }
+
+                if (stat && rtp_session->recv_msg.header.pt != rtp_session->recv_te && rtp_session->recv_msg.header.pt != rtp_session->cng_pt) {
+                    int errs = ++rtp_session->srtp_errs[rtp_session->srtp_idx_rtp];
+                    if (stat != 10) {
+                        char *msg;
+                        if (stat == srtp_err_status_replay_fail) msg="replay check failed (in do_flush)";
+                        else if (stat == srtp_err_status_auth_fail) msg="auth check failed (in do_flush)";
+                        else msg="";
+                        if (errs >= MAX_SRTP_ERRS) {
```

Comment:
we don't hangup anymore by default on srtp errors. see RTP flag SWITCH_RTP_FLAG_SRTP_HANGUP_ON_ERROR